tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -105,6 +120,54 @@ void ROOT::Experimental::RNTupleProcessor::ConnectField(RFieldContext &fieldCont
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RNTupleBaseProcessor::RNTupleBaseProcessor(const RNTupleOpenSpec &ntuple, RNTupleModel &model)
+   : RNTupleProcessor({ntuple})
+{
+   fPageSource = Internal::RPageSource::Create(ntuple.fNTupleName, ntuple.fStorage);
+   fPageSource->Attach();
+
+   if (fPageSource->GetNEntries() == 0) {
+      throw RException(R__FAIL("specified RNTuple \"" + ntuple.fNTupleName + "\" at path \"" + ntuple.fStorage +
+                               "\" does not contain any entries"));
+   }
```

Comment:
For the purposes of this PR I believe we can be restrictive and error out in case the input RNTuple is empty. In general, I would add a bit of information from experience here. With TTree-based processing, cases such as this are possible and fairly common. For example in distributed RDataFrame we create no-op tasks in case we detect an empty TTree in the current task. For RNTuple in general probably now this case is not available or not thoroughly designed, but I would suggest to add a comment here to remember this in the future

Reply:
See  https://github.com/root-project/root/pull/17134#issuecomment-2538516397